packages/core/getUserAvailability.ts

```diff
@@ -626,3 +647,143 @@ const _getBusyTimesFromDurationLimits = async (
     }
   }
 };
+
+interface GetUserAvailabilityParamsDTO {
+  userId: number;
+  dateFrom: Dayjs;
+  dateTo: Dayjs;
+  availability: (DateOverride | WorkingHours)[];
+}
+
+export interface IFromUser {
+  id: number;
+  displayName: string | null;
+}
+
+export interface IToUser {
+  id: number;
+  username: string | null;
+  displayName: string | null;
+}
+
+export interface IOutOfOfficeData {
+  [key: string]: {
+    fromUser: IFromUser | null;
+    toUser?: IToUser | null;
+    reason?: string | null;
+    emoji?: string | null;
+  };
+}
+
+const getOutOfOfficeDays = async (
+  ...args: Parameters<typeof _getOutOfOfficeDays>
+): Promise<ReturnType<typeof _getOutOfOfficeDays>> => {
+  return monitorCallbackAsync(_getOutOfOfficeDays, ...args);
+};
+
+const _getOutOfOfficeDays = async ({
+  userId,
+  dateFrom,
+  dateTo,
+  availability,
+}: GetUserAvailabilityParamsDTO): Promise<IOutOfOfficeData> => {
+  const outOfOfficeDays = await prisma.outOfOfficeEntry.findMany({
+    where: {
+      userId,
+      OR: [
+        // outside of range
+        // (start <= 'dateTo' AND end >= 'dateFrom')
+        {
+          start: {
+            lte: dateTo.toISOString(),
+          },
+          end: {
+            gte: dateFrom.toISOString(),
+          },
+        },
```

Comment:
Shouldn't this alone already cover everything?

Reply:
I identified 3 cases, for a ooo range:
- Can be outside of those two dates but have dates in between.
- Can be between two dates (startMonth, endMonth)
- ooo.rangeStart is before startMonth but ooo.rangeEnd is before endMonth